internal/jimm/access.go

```diff
@@ -807,3 +807,58 @@ func (j *JIMM) ListGroups(ctx context.Context, user *openfga.User, filter pagina
 	}
 	return groups, nil
 }
+
+// OpenFGACleanup queries OpenFGA for all existing tuples, tries to resolve each tuple and removes those
+// that JIMM cannot resolved - orphaned tuples. JIMM not being able to resolve a tuple means that the
+// corresponding entity has been removed from JIMM's database.
+func (j *JIMM) OpenFGACleanup(ctx context.Context) error {
+	for {
+		tuples, continuationToken, err := j.OpenFGAClient.ReadRelatedObjects(ctx, openfga.Tuple{}, 10, "")
+		if err != nil {
+			zapctx.Error(ctx, "reading all tuples", zap.Error(err))
+			return err
+		}
+		orphanedTuples := j.orphanedTuples(ctx, tuples...)
+		if len(orphanedTuples) == 0 {
+			if continuationToken == "" {
+				zapctx.Error(ctx, "continue cleanup")
+				continue
```

Comment:
if you are getting the last tuples and you have no orphaned tuples you continue and requery the tuples from the start. Right?

in a case of cleaned db you are potentially spinning and requery continuously and not once every 6hours

Reply:
fixed, thanks